client/src/lua/mod.rs

```diff
@@ -85,16 +81,47 @@ fn load_shims(lua: rlua::Context) {
     }
 }
 
-/// Sets up the Lua environment for the user code.
-///
-/// This environment is also necessary for some Wayland callbacks,
-/// so it should be called as soon as possible.
-pub fn init_awesome_libraries(lib_paths: &[&str]) {
-    info!("Setting up Awesome libraries");
-    LUA.with(|lua| {
-        lua.borrow()
-            .context(|ctx| register_libraries(ctx, lib_paths).expect("Could not register lua libraries"));
-    });
+pub struct WaylandHandler {}
```

Comment:
`WaylandCallbacks`?  `WaylandAdapter`? `WaylandEventHandlers`?

Reply:
FYI you can drop the `{}` and just put `pub struct WaylandHandler;`. To construct it you simply type `WaylandHandler`. Both should have the same memory representation IIRC (i.e. zero sized type).

I'd prefer `OutputHandler`, since that is what it seems like it is doing? Will we need to centralize _all_ of the handlers under one state structure? It seems better to keep them disjoint, unless you forsee a problem sharing data between them.

Reply:
I don't think that the handlers must share the same instance, let's keep them separate until the need arises (if ever)